### PR TITLE
Add ESLint, fix styling issues

### PR DESCRIPTION
**Pull request by frapontillo**
This PR introduces ESLint, an EcmaScript linter.

The configuration relies on standards from: Node, ES6. It is capable of understanding Jasmine and Promise-specific APIs thanks to custom plugins.

I have added configurations to match our style guidelines:
* semi-colons **everywhere**
* single quotes **everywhere**

The only files you may want to look at if this is the first time you use ESLint are:
* `package.json` where I added the dependencies and changed the `test` task
* `eslintrc.json` where the ESLint configuration sits

All code changes were needed to make the build succeed: for this reason lots of unused variables have been deleted, semi-colons have been added, formatting has been adjusted.

### CODE REVIEW

File: node-server/server/dashboard.js [hunk 1]
```diff
@@ -1,16 +1,3 @@
-const CIWall = require('./widget/ci-wall');
```
Comment (frapontillo) [inline node-server/server/dashboard.js]: These weren't used at all.

File: node-server/server/dashboards.js [hunk 2]
```diff
@@ -1,5 +1,3 @@
-const Dashboard = require('./dashboard.js');
```
Comment (frapontillo) [inline node-server/server/dashboards.js]: Unused.

File: node-server/server/server.spec.js [hunk 3]
```diff
@@ -62,7 +64,7 @@ describe('Server', () => {
     spyOn(dashboards, 'start').and.callThrough();
 
     server.start(io, port);
-    connectClient = listener.on.calls.argsFor(0)[1];
+    const connectClient = listener.on.calls.argsFor(0)[1];
```
Comment (frapontillo) [inline node-server/server/server.spec.js]: We should always declare variables/constants before using them.

File: node-server/server/widget-loader.spec.js [hunk 4]
```diff
@@ -1,10 +1,8 @@
 const WidgetLoader = require('./widget-loader.js');
 const ClassLoader = require('./class-loader.js');
-const Coverage = require('./widget/coverage.js');
 const Error = require('./widget/error.js');
 const fs = require('fs');
-const RightWidget = class RightWidget {}
-const WrongWidget = class WrongWidget {}
```
Comment (frapontillo) [inline node-server/server/widget-loader.spec.js]: This was unused.

File: node-server/server/widget/people-wall.js [hunk 5]
```diff
@@ -48,9 +48,10 @@ class PeopleWall extends BaseWidget {
           return {
             image: image.attribs.src,
             name: image.attribs.title
-          }
-        }).filter(excludeFilter);
-    }
+          };
+        })
+        .filter(excludeFilter);
```
Comment (frapontillo) [inline node-server/server/widget/people-wall.js]: This was not an error but all operators should be applied on a new line for consistency :)

File: node-server/package.json [hunk 6]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "node index.js",
-    "test": "jasmine JASMINE_CONFIG_PATH=jasmine.json"
+    "test": "./node_modules/.bin/eslint server && jasmine JASMINE_CONFIG_PATH=jasmine.json"
```
Comment (florianmski) [inline node-server/package.json]: I think you should be able to use `eslint` here as I did for `jasmine`

**Comment by florianmski** [comment]
Awesome ✨ 
Thanks @frapontillo, just a small Q and then insta-merge ❤️

File: node-server/package.json (hunk 6 above)
Comment (frapontillo) [inline node-server/package.json]: Done in fb4c59efa89094e052b62d0390917bf5d7324197!

**Pull request merged**